soundcard/pulseaudio.py

```diff
@@ -386,19 +386,29 @@ def _match_soundcard(id, soundcards, include_loopback=False):
     else:
         soundcards_by_id = {soundcard['id']: soundcard for soundcard in soundcards}
         soundcards_by_name = {soundcard['name']: soundcard for soundcard in soundcards}
+
+    # index search
+    try:
+        return soundcards[int(id)]
+    except (ValueError):
+        pass
```

Comment:
This block should probably be something like `if int(id) in soundcards` instead of a `try`/`catch`.

Reply:
Yeah, but that adds a lot more complexity. Like I said I have to change a few thing around first. The current source and sink info doesn't even have IDs (afiak anyway). This would take much more work to first update the info obtained from CFFI and then construct a new soundcards with proper info.

This was just a simply one stop shop hack I put together to test my project.

This should be done for readability and future maintenance however. Gimme the go ahead and I'll try and fix it up once and for all if I get some time this week.

Reply:
Absolutely go ahead and investigate it fully. But take it easy; there's absolutely no rush, so do it on your own time.

Let me know when you think it's ready!